src/som/compiler/MixinDefinition.java

```diff
@@ -367,7 +367,7 @@ protected boolean hasOnlyImmutableFields(final HashSet<SlotDefinition> instanceS
           }
         }
 
-        SInitializer mixinInit = cdef.assembleMixinInitializer(i);
```

Comment:
I suspect the for loop above can be changed to be a for each loop over `mixins`, we don't need the `i` anymore,  I think.